filter/store.go

```diff
@@ -80,13 +94,44 @@ func (store *Store) PutCSVData(filterJobID string, fileURL string, size int64) e
 		},
 	}
 
-	json, err := json.Marshal(putBody)
+	return store.updateFilterOutput(filterJobID, &putBody)
+}
+
+// PutStateAsEmpty sets the filter output as empty
+func (store *Store) PutStateAsEmpty(filterJobID string) error {
+	putBody := FilterOuput{
+		State: "completed",
+		Events: Events{
+			Info: []EventItem{EventItem{Message: "No results where found when using the provided filter options",
+				Time: time.Now().UTC()}},
+		},
+	}
+
+	return store.updateFilterOutput(filterJobID, &putBody)
+}
+
+// PutStateAsError set the filter output as an error, we shouldn't state the type of error as
+// this will be displayed to the public
+func (store *Store) PutStateAsError(filterJobID string) error {
+	putBody := FilterOuput{
+		State: "failed",
+		Events: Events{
+			Error: []EventItem{EventItem{Message: "Failed to find any data with the requested information",
+				Time: time.Now().UTC()}},
+		},
+	}
+
+	return store.updateFilterOutput(filterJobID, &putBody)
+}
+
+func (store *Store) updateFilterOutput(filterJobID string, filter *FilterOuput) error {
+	url := store.filterAPIURL + "/filter-outputs/" + filterJobID
+	json, err := json.Marshal(filter)
 	if err != nil {
 		return err
 	}
-
+	log.Debug("json", log.Data{"data": string(json)})
```

Comment:
Is this log line meant to be here? If yes then message should be more descriptive instead of just "json"